packages/tuono-router/src/components/RouterProvider.tsx

```diff
@@ -1,51 +1,32 @@
-import React from 'react'
-import type { ReactNode, JSX } from 'react'
+import type { JSX } from 'react'
+import { Suspense } from 'react'
 
-import { useListenBrowserUrlUpdates } from '../hooks/useListenBrowserUrlUpdates'
-import { useInitRouterStore } from '../hooks/useRouterStore'
 import type { ServerProps } from '../types'
 import type { Router } from '../router'
 
-import { getRouterContext } from './RouterContext'
+import { RouterContextProvider } from './RouterContext'
 import { Matches } from './Matches'
 
-interface RouterContextProviderProps {
-  router: Router
-  children: ReactNode
-}
-
-function RouterContextProvider({
-  router,
-  children,
-}: RouterContextProviderProps): JSX.Element {
-  // Allow the router to update options on the router instance
-  router.update({ ...router.options } as Parameters<typeof router.update>[0])
-
-  const routerContext = getRouterContext()
-
-  return (
-    <React.Suspense>
-      <routerContext.Provider value={router}>{children}</routerContext.Provider>
-    </React.Suspense>
-  )
-}
-
 interface RouterProviderProps {
   router: Router
   serverProps?: ServerProps
 }
 
+/**
+ * This component is the tuono app entry point
```

Comment:
Just a small reword, otherwise it look like that we are targeting this file from the tuono build
```suggestion
 * This component is used in the tuono app entry point
```